src/stateful_check/core.clj

```diff
@@ -131,7 +132,22 @@
                (update try-result :result-data
                        merge {:commands commands, :options options, :specification spec})))))))))
 
-(defn- print-sequence [commands stacktrace? messages]
+(defn- print-failures [handle failures]
+  (doseq [{:keys [message events]} (get failures handle)]
+    (if-not (seq events)
+      (println "   " message)
+      (doseq [{:keys [message] :as event} events]
```

Comment:
I looked into whether we should get away with calling `clojure.test/report` here, instead of our own printing logic. It doesn't play nice with [Leiningen](https://codeberg.org/leiningen/leiningen/src/branch/main/src/leiningen/test.clj#L116-L129), but given [they recommend not using their default test task](https://codeberg.org/leiningen/leiningen/src/branch/main/src/leiningen/test.clj#L13-L23) we might be able to just make that a requirement of using `is` in postconditions.

It would be nice if we could figure this out, because it would interact better with things like `human-test-output`.

I played around with it in a CIDER session, and it reported things in a way that I didn't expect, so there might be more to this than I think.

Reply:
Yeah, not sure about this one. I also played a bit around with the reporting and at one point tried to emit my own events, one for the first failing case and one for the shrunk case and implemented a report function that prints the events. And I believe since everyone is hooking into the report function there would also be one needed for CIDER on the NREPL side. My experiments weren't successful either.